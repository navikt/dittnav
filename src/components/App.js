import React, { Component } from 'react';
import PersonInfo from './PersonInfo';
import InfoMeldinger from './InfoMeldinger';
import Tjenester from './Tjenester';
import Lenkelister from './Lenkelister';
import Artikler from './Artikler';

// // dittnav\web\src\main\java\no\nav\sbl\dialogarena\dittnav\web\pages\ForsideInnlogget.html
class App extends Component {
  render() {
    return (
      <main role="main">
        <section className="service-notification" style={{ display: 'block' }}>
          <h1 className="typo-element">
            <span style={{ display: 'inline' }}>Det skjedde en feil under henting av saker</span>
            <span style={{ display: 'inline' }}>Det skjedde en feil under henting av meldinger fra din innboks</span>
          </h1>
        </section>
        <div className="container">
          <div className="row">
            <div className="maincontent side-innhold">
              <div className="col-md-12">
                <PersonInfo />
                <InfoMeldinger />
                <Tjenester />
                <Lenkelister />
              </div>
            </div>
          </div>

          <Artikler />

        </div>
      </main>
    );
  }
}

export default App;
