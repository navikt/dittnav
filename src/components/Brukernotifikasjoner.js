import React from 'react';
import { bool } from 'prop-types';
import Beskjed from './brukernotifikasjoner/Beskjed';
import Oppgave from './brukernotifikasjoner/Oppgave';
import Innboks from './brukernotifikasjoner/Innboks';
import InnloggingsModal from './common/InnloggingsModal';
import useStore from '../hooks/useStore';
import { byEventTidspunkt } from '../utils/datoUtils';
import useInnloggingsstatus from '../hooks/useInnloggingsstatus';
import BeskjedType from '../types/BeskjedType';
import OppgaveType from '../types/OppgaveType';
import InnboksType from '../types/InnboksType';

const Brukernotifikasjoner = ({ beskjeder, oppgaver, innbokser, erAktiv, erInaktiv }) => {
  const [{ data: innloggingsstatus, isSuccess }] = useInnloggingsstatus();

  if (useStore().state.visInnloggingsModal) {
    return (
      <InnloggingsModal isOpen onClick={() => null} />
    );
  }

  if (!isSuccess) {
    return null;
  }

  return (
    <>
      {oppgaver && innloggingsstatus && oppgaver.content.sort(byEventTidspunkt)
        .map(o => (
          <Oppgave
            key={o.eventId}
            oppgave={o}
            innloggingsstatus={innloggingsstatus.content}
          />
        ))}
      {beskjeder && beskjeder.content && innloggingsstatus && beskjeder.content.sort(byEventTidspunkt)
        .map(b => (
          <Beskjed
            key={b.uid}
            beskjed={b}
            beskjeder={b}
            innloggingsstatus={innloggingsstatus.content}
            erAktiv={erAktiv}
            erInaktiv={erInaktiv}
          />
        ))}
      {innbokser && innloggingsstatus && (
        <Innboks innbokser={innbokser && innbokser.content} innloggingsstatus={innloggingsstatus.content} />
      )}
    </>
  );
};

Brukernotifikasjoner.propTypes = {
  beskjeder: BeskjedType,
  oppgaver: OppgaveType,
  innbokser: InnboksType,
  erAktiv: bool,
  erInaktiv: bool,
};

Brukernotifikasjoner.defaultProps = {
  beskjeder: null,
  oppgaver: null,
  innbokser: null,
  erAktiv: false,
  erInaktiv: false,
};

export default Brukernotifikasjoner;
