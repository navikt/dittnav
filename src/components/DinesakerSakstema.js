import React from 'react';
import { FormattedDate, FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Undertekst, Undertittel } from 'nav-frontend-typografi';
import Lenke from 'nav-frontend-lenker';
import { lenker } from '../utils/lenker';
import { Format } from '../constants';

const sakstemaUrlOverride = {
  KOM: lenker.digisos.url,
  DAG: lenker.dagpenger.url,
  HJE: lenker.hjelpemidler.url,
};

const DinesakerSakstema = (props) => {
  const { sisteOppdatering, temanavn } = props.tema;

  const getTemaUrl = () => {
    const { temakode } = props.tema;
    return sakstemaUrlOverride[temakode] || `${props.url}/tema/${temakode}`;
  };

  return (
    <div className="sak-container">
      <Lenke
        href={getTemaUrl()}
        className="sak-lenke"
        id="sak-lenke-id"
      >
        <div className="sak-temanavn lenke">
          <Undertittel>
            {temanavn}
          </Undertittel>
        </div>

        <div className="sak-status">
          <Undertekst>
            <FormattedMessage id="sakstema.sist.oppdatert" />
            {
              sisteOppdatering && sisteOppdatering !== ''
                ? (
                  <FormattedDate
                    value={moment(sisteOppdatering, Format.SAKSTEMA)}
                    year="numeric"
                    month="short"
                    day="numeric"
                  />
                ) : <FormattedMessage id="sakstema.dato.ukjent" />
            }
          </Undertekst>
        </div>
      </Lenke>
    </div>
  );
};

DinesakerSakstema.propTypes = {
  tema: PropTypes.shape({
    kode: PropTypes.string.isRequired,
    navn: PropTypes.string.isRequired,
    sistEndret: PropTypes.string.isRequired,
  }).isRequired,
  url: PropTypes.string.isRequired,
};

export default DinesakerSakstema;
