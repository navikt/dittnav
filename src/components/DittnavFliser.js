import React from 'react';
import { FormattedMessage as F } from 'react-intl';
import LenkepanelMedIkon from './common/LenkepanelMedIkon';
import PanelOverskrift from './common/PanelOverskrift';
import IkonBlyant from '../assets/IkonBlyant';
import IkonSparegris from '../assets/IkonKane';
import IkonPlaster from '../assets/IkonPlaster';
import IkonSkilt from '../assets/IkonSkilt';
import { GoogleAnalyticsAction, GoogleAnalyticsCategory } from '../utils/googleAnalytics';
import { lenker } from '../utils/lenker';

const DittnavFliser = () => (
  <>
    <div className="dittnav-lenkeikon-container">
      <LenkepanelMedIkon
        alt="Ditt sykefravÃ¦r"
        overskrift={<PanelOverskrift overskrift={<F id="fliser.ditt.sykevravaer" />} type="Undertittel" />}
        ingress={<F id="fliser.ditt.sykevravaer.ingress" />}
        className="first"
        href={lenker.dittSykefravaer.url}
        gaCategory={GoogleAnalyticsCategory.Forside}
        gaAction={GoogleAnalyticsAction.DittSykefravaer}
      >
        <IkonPlaster />
      </LenkepanelMedIkon>
      <LenkepanelMedIkon
        alt="Mistet jobben?"
        overskrift={<PanelOverskrift overskrift={<F id="fliser.mistet.jobben" />} type="Undertittel" />}
        ingress={<F id="fliser.mistet.jobben.ingress" />}
        href={lenker.veilederArbeidssoker.url}
        gaCategory={GoogleAnalyticsCategory.Forside}
        gaAction={GoogleAnalyticsAction.MistetJobben}
      >
        <IkonSkilt />
      </LenkepanelMedIkon>
    </div>
    <div className="dittnav-lenkeikon-container blokk-xxl">
      <LenkepanelMedIkon
        alt="Skjemaer"
        overskrift={<PanelOverskrift overskrift={<F id="fliser.skjemaer" />} type="Undertittel" />}
        ingress={<F id="fliser.skjemaer.ingress" />}
        className="first"
        href={lenker.skjemaer.url}
        gaCategory={GoogleAnalyticsCategory.Forside}
        gaAction={GoogleAnalyticsAction.Skjemaer}
      >
        <IkonBlyant />
      </LenkepanelMedIkon>
      <LenkepanelMedIkon
        alt="Din pensjon"
        overskrift={<PanelOverskrift overskrift={<F id="fliser.din.pensjon" />} type="Undertittel" />}
        ingress={<F id="fliser.din.pensjon.ingress" />}
        href={lenker.dinPensjon.url}
        gaCategory={GoogleAnalyticsCategory.Forside}
        gaAction={GoogleAnalyticsAction.DinPensjon}
      >
        <IkonSparegris />
      </LenkepanelMedIkon>
    </div>
  </>
);

export default DittnavFliser;
