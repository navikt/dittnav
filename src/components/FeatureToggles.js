import React, { useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Path, Toggle } from '../constants';

export const FeatureToggles = React.createContext({});

export const FeatureToggleWrapper = ({ toggle, children }) => {
  const { featureToggles } = useContext(FeatureToggles);
  return (
    <>{featureToggles != null
      ? React.cloneElement(children, { isFeatureEnabled: featureToggles[toggle] })
      : React.cloneElement(children, { isFeatureEnabled: false })}
    </>
  );
};

FeatureToggleWrapper.propTypes = {
  toggle: PropTypes.string,
  children: PropTypes.node,
};

FeatureToggleWrapper.defaultProps = {
  toggle: '',
  children: {},
};

const FeatureTogglesProvider = ({ children }) => {
  // const loaded = false;
  // const featureToggles = {};
  /*
  TODO: Utkommentert i påvente av at vi skriver vår egen unleash-toggle-provider istedenfor å bruke Unleash gjennom pus-dekoratør.
  */
  const [featureToggles, setFeatureToggles] = useState({});
  const [loaded, setLoaded] = useState(false);

  const UNLEASH_TIMEOUT = 1000;

  useEffect(() => {
    const createURL = () => {
      const toggles = Toggle.FEATURE_TOGGLES.split(',');
      const togglePath = toggles.reduce((accumulatedToggles, currentToggle) => `${accumulatedToggles}&feature=${currentToggle}`);
      return `${Path.CONTEXT}/api/feature?feature=${togglePath}`;
    };
    Promise.race([
      fetch(createURL(), { method: 'GET' })
        .then(r => r.json()),
      new Promise((_, reject) => setTimeout(() => {
        const message = `Couldnt wait for unleash longer than ${UNLEASH_TIMEOUT} msec`;
        return reject(new Error(message));
      }, UNLEASH_TIMEOUT))])
      .then(response => setFeatureToggles(response))
      .catch(err => {
        // eslint-disable-next-line no-console
        console.log(err);
      })
      .finally(() => setLoaded(true));
  }, []);

  if (loaded) {
    return (
      <FeatureToggles.Provider value={{ featureToggles }}>
        {children}
      </FeatureToggles.Provider>
    );
  } return null;
};

FeatureTogglesProvider.propTypes = {
  children: PropTypes.node,
};

FeatureTogglesProvider.defaultProps = {
  children: {},
};

export default FeatureTogglesProvider;
