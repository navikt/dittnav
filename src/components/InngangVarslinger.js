import React from 'react';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';
import { Undertittel } from 'nav-frontend-typografi';
import HoyreChevron from 'nav-frontend-chevron';
import { listOfActions, listOfComponentNames, logAmplitudeEvent } from '../utils/amplitudeUtils';

const InngangVarslinger = () => (
  <div className="varslinger-inngang-wrapper">
    <Undertittel className="varslinger-inngang">
      <Link
        id="varslinger-inngang__lenke-id"
        to="/dittnav/varslinger"
        onClick={() => logAmplitudeEvent(listOfComponentNames.brukernotifikasjon.TidligereBeskjederOgOppgaver, listOfActions.TrykkPaaLenke)}
      >
        <FormattedMessage id="dittnav.infomeldinger.inngang.varslinger" />
      </Link>
      <HoyreChevron className="varslinger-inngang__chevron" />
    </Undertittel>
  </div>
);

export default InngangVarslinger;
