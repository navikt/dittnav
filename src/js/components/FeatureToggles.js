import React, { useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';
import Config from '../Config';

export const FeatureToggles = React.createContext({});

export const FeatureToggleWrapper = ({ toggle, children }) => {
  const { featureToggles } = useContext(FeatureToggles);
  return (
    <>{featureToggles != null
      ? React.cloneElement(children, { isFeatureEnabled: featureToggles[toggle] })
      : React.cloneElement(children, { isFeatureEnabled: false })}
    </>
  );
};

FeatureToggleWrapper.propTypes = {
  toggle: PropTypes.string,
  children: PropTypes.node,
};

FeatureToggleWrapper.defaultProps = {
  toggle: '',
  children: {},
};

const FeatureTogglesProvider = ({ children }) => {
  const [featureToggles, setFeatureToggles] = useState({});

  useEffect(() => {
    const createURL = () => {
      const toggles = Config.dittNav.FEATURE_TOGGLES.split(',');
      const togglePath = toggles.reduce((accumulatedToggles, currentToggle) => `${accumulatedToggles}&feature=${currentToggle}`);
      return `${Config.dittNav.CONTEXT_PATH}/api/feature?feature=${togglePath}`;
    };

    fetch(createURL(), { method: 'GET' })
      .then(r => r.json())
      .then(response => setFeatureToggles(response))
      // eslint-disable-next-line no-console
      .catch(err => console.log(err));
  }, []);

  return (
    <FeatureToggles.Provider value={{ featureToggles }}>
      {children}
    </FeatureToggles.Provider>
  );
};

FeatureTogglesProvider.propTypes = {
  children: PropTypes.node,
};

FeatureTogglesProvider.defaultProps = {
  children: {},
};

export default FeatureTogglesProvider;
