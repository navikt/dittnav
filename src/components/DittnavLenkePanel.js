import React from 'react';
import { FormattedMessage as F } from 'react-intl';
import Lenkepanel from 'nav-frontend-lenkepanel/lib';
import OversiktspanelMedListe from './common/OversiktspanelMedListe';
import DinesakerSakstema from './DinesakerSakstema';
import { listOfActions, listOfComponentNames, logAmplitudeEvent } from '../utils/amplitudeUtils';
import { lenker } from '../utils/lenker';
import { useSakstema } from '../hooks/useSaker';

const antallSakstemaVist = 2;

const DittnavLenkePanel = () => {
  const [{ data: saker }] = useSakstema();
  const visStortSakspanel = saker && saker.content.sakstemaer?.length > 0;
  const sakerURL = saker?.content?.sakerURL;

  console.log(sakerURL)

  return (
    <div className="dittnav-lenkepanel-top-container">
      {visStortSakspanel
        ? (
          <OversiktspanelMedListe
            className="dittnav-lenkepanel-stor"
            overskrift={<F id="saksoversikt.overskrift" />}
            headerLenkeTekst={<F id="saksoversikt.alle.saker" />}
            headerLenkeHref={sakerURL}
            border={false}
            liste={
              saker.content.sakstemaer
                .slice(0, antallSakstemaVist)
                .map((sak) => (
                  <DinesakerSakstema
                    key={sak.kode}
                    tema={sak}
                  />
                ))
            }
          />
        ) : null}
      <div className="dittnav-lenkepanel-liten" id={visStortSakspanel ? 'cols-layout' : null}>
        {!visStortSakspanel
          ? (
            <Lenkepanel
              alt="Dine saker"
              className="dittnav-lenkepanel-liten-item"
              onClick={() => logAmplitudeEvent(listOfComponentNames.DineSisteSakerFlis, listOfActions.TrykkPaaLenke)}
              href={saker?.content.sakerURL}
              border
            >
              <F id="fliser.dine.saker" />
            </Lenkepanel>
          ) : null}
        <Lenkepanel
          alt="Utbetalinger"
          className="dittnav-lenkepanel-liten-item"
          href={lenker.utbetalingsoversikt.url}
          onClick={() => logAmplitudeEvent(listOfComponentNames.UtbetalingerFlis, listOfActions.TrykkPaaLenke)}
          border
        >
          <F id="fliser.dine.utbetalinger" />
        </Lenkepanel>
        <Lenkepanel
          alt="Innboks"
          className="dittnav-lenkepanel-liten-item last"
          href={lenker.innboks.url}
          onClick={() => logAmplitudeEvent(listOfComponentNames.InnboksFlis, listOfActions.TrykkPaaLenke)}
          border
        >
          <F id="fliser.innboks" />
        </Lenkepanel>
      </div>
    </div>
  );
};

export default DittnavLenkePanel;
