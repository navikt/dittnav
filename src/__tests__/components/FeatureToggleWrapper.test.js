/* eslint-disable react/prop-types */

import * as React from 'react';
import { FeatureToggleWrapper, FeatureToggles } from '../../js/components/FeatureToggles';

const ReactTestRenderer = require('react-test-renderer');
const C = props => (props.isFeatureEnabled ? <div>Hello, the feature is enabled</div> : <div>Not hello, the feature is disabled</div>);

test('feature is enabled Unleash snaphot-test', async () => {
  const featureToggles = { "dittnav.fo": true };
  const component = ReactTestRenderer.create(
    <FeatureToggles.Provider value={{ featureToggles }}>
      <FeatureToggleWrapper toggle="dittnav.fo">
        <C />
      </FeatureToggleWrapper>
    </FeatureToggles.Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('feature is disabled Unleash snaphot-test', async () => {
  const featureToggles = { "dittnav.fo": false };
  const component = ReactTestRenderer.create(
    <FeatureToggles.Provider value={{ featureToggles }}>
      <FeatureToggleWrapper toggle="dittnav.fo">
        <C />
      </FeatureToggleWrapper>
    </FeatureToggles.Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('no such feature Unleash snaphot-test', async () => {
  const featureToggles = { "dittnav.notfo": true };
  const component = ReactTestRenderer.create(
    <FeatureToggles.Provider value={{ featureToggles }}>
      <FeatureToggleWrapper toggle="dittnav.fo">
        <C />
      </FeatureToggleWrapper>
    </FeatureToggles.Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('no set feature toggles', async () => {
  const component = ReactTestRenderer.create(
    <FeatureToggles.Provider value={{ }}>
      <FeatureToggleWrapper toggle="dittnav.fo">
        <C />
      </FeatureToggleWrapper>
    </FeatureToggles.Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
