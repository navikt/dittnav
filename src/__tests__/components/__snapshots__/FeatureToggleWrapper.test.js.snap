// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature is disabled Unleash snaphot-test 1`] = `
<div>
  Not hello, the feature is disabled
</div>
`;

exports[`feature is enabled Unleash snaphot-test 1`] = `
<div>
  Hello, the feature is enabled
</div>
`;

exports[`no set feature toggles 1`] = `
<div>
  Not hello, the feature is disabled
</div>
`;

exports[`no such feature Unleash snaphot-test 1`] = `
<div>
  Not hello, the feature is disabled
</div>
`;
